app/actions/bonds.ts

```diff
@@ -315,15 +315,15 @@ export const calcBondDetails = (params: {
 export const calculateUserBondDetails = (params: {
   address: string;
   bond: Bond;
-  provider: providers.JsonRpcProvider;
 }): Thunk => {
   return async (dispatch) => {
     if (!params.address) return;
+    const provider = getStaticProvider();
```

Comment:
Can you say again, why you replaced the provider from the useWeb3 hook with the staticProvider again?

Because the contract just needs the "simple" provider?
And to ensure less requests here too?

Reply:
Just because the user RPC endpoint can be unreliable. So we only really want to use that for outgoing txns (as some users might prefer to set up their own RPC for privacy), but even then I am considering *always* using the static provider even for outgoing...